src/reconcile/rec_child.c

```diff
@@ -228,8 +218,8 @@ __wt_rec_child_modify(
              * checkpoint, all splits in process will have completed before we walk any pages for
              * checkpoint.
              */
-            WT_ASSERT(session, WT_REF_SPLIT != WT_REF_SPLIT);
-            return (__wt_set_return(session, EBUSY));
+            WT_ERR_ASSERT(
+              session, false, EBUSY, "unexpected WT_REF_SPLIT child state during reconciliation");
```

Comment:
Needs to be a `WT_REF_ASSERT`: there's no error label.  This causes non-diagnostic builds to fail.

Reply:
Thanks, pushed in 6907fb9de.